wicket-core/src/main/java/org/apache/wicket/pageStore/DiskPageStore.java

```diff
@@ -0,0 +1,717 @@
+			diskData.unbind();
+		}
+	}
+
+	/**
+	 * Supports {@link SerializedPage}s too - for this to work the delegating {@link IPageStore}
+	 * must use the same {@link ISerializer} as this one.
+	 */
+	@Override
+	public void addPage(IPageContext context, IManageablePage page)
+	{
+		DiskData diskData = getDiskData(context, true);
+		if (diskData != null)
+		{
+			log.debug("Storing data for page with id '{}' in session with id '{}'",
+				page.getPageId(), context.getSessionId());
+
+			byte[] data;
+			String type;
+			if (page instanceof SerializedPage)
+			{
+				data = ((SerializedPage)page).getData();
+				type = ((SerializedPage)page).getPageType();
+			}
+			else
+			{
+				if (serializer == null)
+				{
+					throw new WicketRuntimeException(
+						"DiskPageStore not configured for serialization");
+				}
+				data = serializer.serialize(page);
+				type = page.getClass().getName();
+			}
+
+			diskData.savePage(page.getPageId(), type, data);
+		}
+	}
+
+	/**
+	 * 
+	 * @param context
+	 * @param create
+	 * @return the session entry
+	 */
+	protected DiskData getDiskData(final IPageContext context, final boolean create)
+	{
+		SessionAttribute attribute = getSessionAttribute(context, create);
+
+		if (!create && attribute == null)
+		{
+			return null;
+		}
+
+		return getDiskData(attribute.identifier, create);
+	}
+
+	protected DiskData getDiskData(String sessionIdentifier, boolean create)
+	{
+		if (!create)
+		{
+			return diskDatas.get(sessionIdentifier);
+		}
+
+		DiskData data = new DiskData(this, sessionIdentifier);
+		DiskData existing = diskDatas.putIfAbsent(sessionIdentifier, data);
+		return existing != null ? existing : data;
+	}
+
+	protected SessionAttribute getSessionAttribute(IPageContext context, boolean create)
+	{
+		SessionAttribute attribute = context.getSessionAttribute(KEY);
+		if (attribute == null && create)
+		{
+			attribute = new SessionAttribute(applicationName, context.getSessionId());
+			context.setSessionAttribute(KEY, attribute);
+		}
+		return attribute;
+	}
+
+	/**
+	 * Load the index
+	 */
+	@SuppressWarnings("unchecked")
+	private void loadIndex()
+	{
+		File storeFolder = folders.getBase();
+
+		File index = new File(storeFolder, INDEX_FILE_NAME);
+		if (index.exists() && index.length() > 0)
+		{
+			try
+			{
+				InputStream stream = new FileInputStream(index);
+				ObjectInputStream ois = new ObjectInputStream(stream);
+				try
+				{
+					diskDatas.clear();
+
+					for (DiskData diskData : (List<DiskData>)ois.readObject())
+					{
+						diskData.pageStore = this;
+						diskDatas.put(diskData.sessionIdentifier, diskData);
+					}
+				}
+				finally
+				{
+					stream.close();
+					ois.close();
+				}
+			}
+			catch (Exception e)
+			{
+				log.error("Couldn't load DiskPageStore index from file " + index + ".", e);
+			}
+		}
+		Files.remove(index);
+	}
+
+	/**
+	 * 
+	 */
+	private void saveIndex()
+	{
+		File storeFolder = folders.getBase();
+		if (storeFolder.exists())
```

Comment:
Should we try to create it if it does not exist ?

Reply:
At this moment it is created if at least one page was written to disk.

That code comes from the former DiskDataStore, and I've moved all code the folder handling to the NestedFolders utilty,